src/lambda/lambdaPolicy.ts

```diff
@@ -0,0 +1,175 @@
+}
+
+export class DefaultLambdaPolicyProvider implements LambdaPolicyProvider {
+    public functionName: string
+    private readonly _lambdaClient: Lambda
+
+    public constructor(
+        functionName: string,
+        lambdaClient: Lambda
+    ) {
+        if (functionName.length === 0) {
+            throw new Error('Lambda function name is missing')
+        }
+
+        this.functionName = functionName
+        this._lambdaClient = lambdaClient
+    }
+
+    public async getLambdaPolicy(): Promise<Lambda.GetPolicyResponse> {
+        const response = await this._lambdaClient.getPolicy({
+            FunctionName: this.functionName
+        }).promise()
+
+        return response
+    }
+}
+
+export enum LambdaPolicyViewStatus {
+    Initialized,
+    Loading,
+    Loaded,
+    Error,
+    Disposed
+}
+
+export class LambdaPolicyView implements vscode.Disposable {
+
+    private static readonly VIEW_TEMPLATE: _.TemplateExecutor = _.template(BaseTemplates.SIMPLE_HTML)
+    private static readonly VIEW_CONTENT_OUTER_CONTENT: _.TemplateExecutor =
+        _.template(LambdaPolicyTemplates.OUTER_TEMPLATE)
+    private static readonly VIEW_CONTENT_TEMPLATE_LOADING: _.TemplateExecutor =
+        _.template(LambdaPolicyTemplates.INNER_TEMPLATE_LOADING)
+    private static readonly VIEW_CONTENT_TEMPLATE_POLICY: _.TemplateExecutor =
+        _.template(LambdaPolicyTemplates.INNER_TEMPLATE_POLICY)
+    private static readonly VIEW_CONTENT_TEMPLATE_ERROR: _.TemplateExecutor =
+        _.template(LambdaPolicyTemplates.INNER_TEMPLATE_ERROR)
+
+    protected _view: vscode.WebviewPanel | undefined
+
+    private readonly _lambdaPolicyProvider: LambdaPolicyProvider
+    private _status: LambdaPolicyViewStatus
+
+    public constructor(
+        lambdaPolicyProvider: LambdaPolicyProvider
+    ) {
+        this._lambdaPolicyProvider = lambdaPolicyProvider
+
+        this._view = this.createView()
+        this._status = LambdaPolicyViewStatus.Initialized
+    }
+
+    public async load(): Promise<void> {
+        this.showLoadingContent()
+
+        try {
+            const policyResponse: Lambda.GetPolicyResponse = await this._lambdaPolicyProvider.getLambdaPolicy()
+
+            this.showPolicyContent(policyResponse)
+        } catch (err) {
+            console.log('Error loading and showing Lambda Policy', err)
+            this.showError(err as Error)
+        }
+    }
+
+    public getStatus(): LambdaPolicyViewStatus {
+        return this._status
+    }
+
+    public dispose() {
+        if (!!this._view) {
+            this._view.dispose()
+        }
+    }
+
+    private createView(): vscode.WebviewPanel {
+
+        const view: vscode.WebviewPanel = vscode.window.createWebviewPanel(
+            'html',
+            localize(
+                'AWS.lambda.policy.title',
+                'Lambda Policy: {0}',
+                this._lambdaPolicyProvider.functionName
+            ),
+            -1
+        )
+
+        view.onDidDispose(() => {
```

Comment:
Do we also need to unregister the `onDidDispose` handler, or does that happen automatically?

Reply:
Good catch!

"Calls to subscribe return a Disposable which removes the event listener upon dispose."
https://code.visualstudio.com/docs/extensionAPI/patterns-and-principles#_events